### PR TITLE
gh-104242: Skip test_is_char_device_true in pathlib test on non posix platform

### PR DESCRIPTION
**Pull request by aisk**
* Issue: gh-104242

### CODE REVIEW

File: Lib/test/test_pathlib/test_pathlib.py [hunk 1]
```diff
@@ -1102,6 +1102,7 @@ def test_is_socket_true(self):
         self.assertIs(self.cls(self.base, 'mysock\udfff').is_socket(), False)
         self.assertIs(self.cls(self.base, 'mysock\x00').is_socket(), False)
 
+    @unittest.skipIf(os.name != 'posix', 'test requires a POSIX-compatible system')
```
Comment (barneygale) [inline Lib/test/test_pathlib/test_pathlib.py]:
```suggestion
    @needs_posix
```

**Review by barneygale — CHANGES_REQUESTED** [review]

File: Lib/test/test_pathlib/test_pathlib.py (hunk 1 above)
Comment (aisk) [reply Lib/test/test_pathlib/test_pathlib.py]: Thank you for the review! I think the `needs_posix` will skip the test if the current test class is not using a posix flavor path class, and they may run on Windows like the `PurePosixPathTest` test class.

In this scenario, we want to skip the test case if the current platform is not posix, so I think the `needs_posix` is not suitable here.

Comment (barneygale) [reply Lib/test/test_pathlib/test_pathlib.py]: `PathTest.test_is_char_device_true()` runs only against `Path` and its subclasses, not `PurePath`. I do think `@needs_posix` is the right thing to do here, and its consistent with other `PathTest` methods.

**Review by serhiy-storchaka — COMMENTED** [review]
Why not use `os.devnull`?

**Comment by aisk** [comment]
> Why not use `os.devnull`?

Good idea, I'll create another PR with this mechanism for other reviewers to keep a clear review history.

**Pull request closed without merging**